packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -0,0 +1,45 @@
+import { attr } from '@microsoft/fast-element';
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import { template } from '../theme-provider/template';
+import type { WaferMapRenderingObject } from './models/wafer-map-rendering-object';
+import { styles } from './styles';
+import { WaferMapColorBy, WaferMapColorsScale } from './types';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-wafer-map': WaferMap;
+    }
+}
+
+/**
+ * A nimble-styled WaferMap
+ */
+export class WaferMap extends FoundationElement {
+    @attr({
+        attribute: 'waferData'
+    })
+    public waferData!: WaferMapRenderingObject;
```

Comment:
I would also avoid using complex dataTypes (like WaferMapRenderingObject here) as an attribute ... I think that this would make the web-component unnecessarily hard to use. Instead, I would suggest to focus on having simple attributes that could be re-assembled into a more complex type inside the component itself.

Reply:
Done :)